UIforETW/Utility.cpp

```diff
@@ -127,71 +276,125 @@ std::wstring ConvertToCRLF(const std::wstring& input)
 		if (c != '\r')
 			result += c;
 	}
-
 	return result;
 }
 
-void SetRegistryDWORD(HKEY root, const std::wstring& subkey, const std::wstring& valueName, DWORD value)
+void SetRegistryDWORD(const HKEY root, const std::wstring& subkey, const std::wstring& valueName, const DWORD value)
 {
 	HKEY key;
-	LONG result = RegOpenKeyEx(root, subkey.c_str(), 0, KEY_ALL_ACCESS, &key);
-	if (result == ERROR_SUCCESS)
-	{
-		RegSetValueEx(key, valueName.c_str(), 0, REG_DWORD, reinterpret_cast<const BYTE*>(&value), sizeof(value));
-		RegCloseKey(key);
-	}
+	ATLVERIFY(::RegOpenKeyExW(root, subkey.c_str(), 0, KEY_ALL_ACCESS, &key) == ERROR_SUCCESS);

```

Comment:
This changes RegOpenKeyEx from being a run-time error to being a logic error. I'm going to change it back because it's not a logic error.